templates/task-sc.yaml

```diff
@@ -0,0 +1,53 @@
+---
+apiVersion: tekton.dev/v1beta1
+kind: Task
+metadata:
+    name: skopeo-copy
+    labels:
+        app.kubernetes.io/version: 0.0.1
+spec:
+    description: | 
+        Skopeo is a command line tool for working with remote image registries. 
+        The copy command will take care of copying the image from internal.registry to production.registry. If your production registry requires credentials to login in order to push the image, skopeo can handle that as well.
+    params:
+        -   name: SOURCE_REGISTRY
+            type: string
+            description: |
+                Source Registry URL
+        -   name: DESTINATION_REGISTRY
+            type: string
+            description: |
+                Destination Registry URL
+        -   name: SOURCE_AUTH_USERNAME
+            type: string
+            description: |
+                Source Registry Username for authentication
+        -   name: SOURCE_AUTH_PASSWORD
+            type: string
+            description: |
+                Source Registry Password for authentication
+        -   name: DESTINATION_AUTH_USERNAME
+            type: string
+            description: |
+                Destination Registry Username for authentication
+        -   name: DESTINATION_AUTH_PASSWORD
+            type: string
+            description: |
+                Destination Registry Password for authentication
+    steps:
+    -   name: login-src
+        image: quay.io/skopeo/stable:latest
+        script: |
+            skopeo login docker.io -u=$(params.SOURCE_AUTH_USERNAME) -p=$(params.SOURCE_AUTH_PASSWORD)
+    -   name: login-dest
+        image: quay.io/skopeo/stable:latest
+        script: |
+            skopeo login docker.io -u=$(params.DESTINATION_AUTH_USERNAME) -p=$(params.DESTINATION_AUTH_PASSWORD)
+    -   name: copy
+        image: quay.io/skopeo/stable:latest
+        script: |
+            skopeo copy --src-creds $(params.SOURCE_AUTH_USERNAME):$(params.SOURCE_AUTH_PASSWORD) \
+            docker://$(params.SOURCE_REGISTRY) --dest-creds $(params.DESTINATION_AUTH_USERNAME):$(params.DESTINATION_AUTH_PASSWORD) \
```

Comment:
What does `docker://` protocol means for the container image location?

Reply:
I am not sure this task-sc.yaml file is not written by me. It was merged few days ago.

Reply:
It is basically used for registry implementing the "Docker Registry HTTP API V2". So like if we want to work with registries like docker,  quay, federa, etc. All examples I looked upto were using this protocol for specifying the image.